src/ivc/protogalaxy/mod.rs

```diff
@@ -513,6 +523,59 @@ mod verify_chip {
             .map_err(|err| Error::BetasStroke { err })
     }
 
+    /// Evaluate the values of the Lagrange polynomial zero for a cyclic subgroup of length `n` (`2.pow(log_n)`) at
+    /// the `challenge` point
+    ///
+    /// You can look at [`fft::get_omega_or_inv`] to see how the target cyclic group is formed
+    ///
+    /// # Mathematical Representation
+    ///
+    /// ```math
+    /// L_i(X)=\frac{\omega^i}{n}\frac{X^n-1}{X-\omega^i}
+    /// ```
+    /// where {1, \omega, \omega^2, ..., \omega^n} - cyclic group, check [`iter_cyclic_subgroup`] for
+    /// more details
+    ///
+    /// # Generics
+    /// `T` is setup for main gate
+    /// - `L`: 'Length' - constant representing the number of instances to
+    ///                   fold in a single `prove`. `L-1` be power of two
+    fn eval_lagrange_zero_poly<F: PrimeField, const T: usize, const L: usize>(
+        region: &mut RegionCtx<F>,
+        main_gate: &MainGate<F, T>,
+        cha: &mut ValuePowers<F>,
+    ) -> Result<AssignedValue<F>, Halo2PlonkError> {
+        let lagrange_domain = PolyContext::<F>::get_lagrange_domain::<L>();
+        let points_count = 2usize.pow(lagrange_domain);
+
+        let inverted_n = F::from_u128(points_count as u128)
+            .invert()
+            .expect("safe because it's `2^log_n`");
+        let value = iter_cyclic_subgroup::<F>(lagrange_domain).next().unwrap();
+
+        let X = cha.value();
+
+        let X_sub_value = main_gate.add_with_const(region, &X, -value)?;
+
+        let (is_zero_X_sub_value, X_sub_value_inverted) =
+            main_gate.invert_with_flag(region, X_sub_value)?;
+
+        let X_pow_n = cha.get_or_eval(region, main_gate, points_count)?;
+        let X_pow_n_sub_1 = main_gate.add_with_const(region, &X_pow_n, -F::ONE)?;
+
+        let is_zero_X_pow_n_sub_1 = main_gate.is_zero_term(region, X_pow_n_sub_1.clone())?;
+
+        let is_numerator_denominator_zero =
+            main_gate.mul(region, &is_zero_X_sub_value, &is_zero_X_pow_n_sub_1)?;
+
+        let lhs = main_gate.mul(region, &X_pow_n_sub_1, &X_sub_value_inverted)?;
+        let fractional = main_gate.mul_by_const(region, &lhs, value * inverted_n)?;
+
+        let one = cha.get_or_eval(region, main_gate, 0)?;
```

Comment:
Does this ensure the 0-th power is 1? It is not obvious to me that how the initial value is defined here

Reply:
We even pass a assigned version of `one` to constructor

In general, it seems obvious to me that if you take a degree function and request `0`, it should return one.

However, if you want, I can specify this separately in the rustdoc of this function if you want to

Reply:
no, it looks good. just want to confirm it.